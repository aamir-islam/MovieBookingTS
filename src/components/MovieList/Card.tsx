import React from "react";
import {
  CardStyle,
  CardsStyle,
  ButtonStyle,
  TitleStyle,
  CardsImageStyle,
} from "../Styles/Card.style";
import { fakeImg, imgUrl } from "../constants/global";

import { NavLink } from "react-router-dom";

type Props = {
  data: ApiData[];
};
type ApiData = {
  poster_path: string;
  title: string;
  id: number;
};

const Card = ({ data }: Props) => {
  const SetImagePath = (title:string,path:string)=>{
    localStorage.setItem(title,path)
  }
  return (
    <CardsStyle>
      {data.map((item: ApiData, index) => {
        return (
          <CardStyle key={index}>
            <CardsImageStyle src={item.poster_path ? imgUrl + item.poster_path : fakeImg} alt="path" />
            <TitleStyle>{item.title}</TitleStyle>
            <NavLink to={`/booking/${item.id}/${item.title}`}>
              <ButtonStyle onClick={()=>SetImagePath(item.title,item.poster_path)}>Book Now</ButtonStyle>
            </NavLink>
          </CardStyle>
        );
      })}
    </CardsStyle>
  );
};

export default Card;
