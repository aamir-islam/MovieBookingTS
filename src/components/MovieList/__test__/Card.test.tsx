import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import Card from "../Card";
import { BrowserRouter } from "react-router-dom";

const data = [
  {
    id: 507086,
    poster_path: "/kAVRgw7GgK1CfYEJq8ME6EvRIgU.jpg",
    title: "Jurassic World Dominion",
  },
];
describe("testing Movie Cards", () => {
  it('should take a snapshot', () => {
    const { container } = render(<BrowserRouter><Card responseData={data} /></BrowserRouter>);
    expect(container).toMatchSnapshot()
  })
  it("image should be in Dom", () => {
    render(<BrowserRouter><Card responseData={data} /></BrowserRouter>);
    const element = screen.getByRole<HTMLImageElement>("img");
    expect(element).toBeInTheDocument();
  })
  it("heading should be in Dom", () => {
    render(<BrowserRouter><Card responseData={data} /></BrowserRouter>);
    const elementAlt = screen.getByText("Jurassic World Dominion");
    expect(elementAlt).toBeInTheDocument();
  })

  it("NavLink should have id and path attribute", () => {
    render(<BrowserRouter><Card responseData={data} /></BrowserRouter>);
    const Nav = screen.getByRole<HTMLAnchorElement>("link");
    expect(Nav).toHaveAttribute("href", "/booking/507086/Jurassic World Dominion")
  })
  it("button click", () => {
    render(<BrowserRouter><Card responseData={data} /></BrowserRouter>);
    const primaryButton = screen.getByRole('button', { name: /Book Now/i })
    fireEvent.click(primaryButton);
  })

})

