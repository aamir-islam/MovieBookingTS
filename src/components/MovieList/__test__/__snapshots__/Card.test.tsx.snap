// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Movie Cards should take a snapshot 1`] = `
<div>
  <article
    class="sc-gKXOVf fYbSOE"
  >
    <article
      class="sc-ftvSup irwBlD"
    >
      <img
        alt="path"
        class="sc-iBkjds bUgpid"
        src="https://image.tmdb.org/t/p/w500//kAVRgw7GgK1CfYEJq8ME6EvRIgU.jpg"
      />
      <h1
        class="sc-eCYdqJ gbMNTl"
      >
        Jurassic World Dominion
      </h1>
      <a
        class=""
        href="/booking/507086/Jurassic World Dominion"
      >
        <button
          class="sc-bczRLJ fPpixa"
          data-testid="closeFunc"
        >
          Book Now
        </button>
      </a>
    </article>
  </article>
</div>
`;
